framework/src/meshgenerators/SubdomainBoundingBoxGenerator.C

```diff
@@ -55,19 +57,37 @@ SubdomainBoundingBoxGenerator::generate()
 {
   std::unique_ptr<MeshBase> mesh = std::move(_input);
 
-  // Loop over the elements
-  for (const auto & elem : mesh->active_element_ptr_range())
+  if (isParamValid("integer_name"))
   {
-    bool contains = _bounding_box.contains_point(elem->centroid());
-    if (contains && _location == "INSIDE")
-      elem->subdomain_id() = _block_id;
-    else if (!contains && _location == "OUTSIDE")
-      elem->subdomain_id() = _block_id;
+    std::string integer_name = getParam<std::string>("integer_name");
+
+    if (!mesh->has_elem_integer(integer_name))
+      mooseError("Mesh does not have an element integer names as ", integer_name);
+
+    unsigned int id = mesh->get_elem_integer_index(integer_name);
```

Comment:
This should use the interface... right?

Reply:
No with my second thought. I initially planed to use the interface, but the mesh generators get executed much early when the FEProblems, Assembly, stuff are not available. I think using the libMesh interface functions should be fine in mesh generators.

Reply:
Ok